swagger.yaml

```diff
@@ -75,29 +185,53 @@ paths:
           schema:
             $ref: '#/definitions/Datasets'
         500:
-          description: "Failed to process the request due to an internal error"
-
+          $ref: '#/responses/InternalError'
   /datasets/{id}:
     get:
       tags:
-      - "Dataset"
+      - "Public"
       summary: "Get a dataset"
-      description: "The dataset contains all high level information, for additional details see editions or versions of a dataset"
+      description: "The dataset contains all high level information, for additional details see editions or versions of a dataset. For unpublished datasets a privater user can access them but a public user will be returned a 403 forbidden code"
       parameters:
       - $ref: '#/parameters/id'
       responses:
         200:
           description: "A json object for a single Dataset"
           schema:
-            $ref: '#/definitions/Dataset'
+            $ref: '#/definitions/DatasetResponse'
+        401:
+          description: "The token provided is unauthorised to carry out this operation"
+        403:
```

Comment:
A public user will never get forbidden or unauthorised to view a dataset document. They will get not found....it doesn't exist in their world view.

If there is a dataset which is not public yet when they call this endpoint not found is returned.

Reply:
If a private user puts in the wrong authentication they get the same responses as a public user. If they put the correct authentication header in then they may get more data back.

Reply:
The above 2 comments can be applied to all GET endpoints for dataset, edition and version resources